app/modules/code_provider/code_provider_controller.py

```
@@ -250,15 +331,28 @@ async def get_branch_list(
                 detail=f"Repository {repo_name} not found or error fetching branches: {str(e)}",
             )
 
-    async def get_user_repos(self, user: Dict[str, Any]) -> Dict[str, Any]:
+    async def get_user_repos(self, user: Dict[str, Any], search: str = None) -> Dict[str, Any]:
```

Ruff (0.15.0)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)